native/cocos/core/Root.cpp

```diff
@@ -399,6 +442,50 @@ void Root::destroyWindows() {
     _windows.clear();
 }
 
+uint32_t Root::createSystemWindow(const ISystemWindowInfo &info) {
+    auto *windowMgr = CC_GET_PLATFORM_INTERFACE(ISystemWindowManager);
+    ISystemWindow *window = windowMgr->createWindow(info);
+    if (window) {
+        auto *screen = CC_GET_PLATFORM_INTERFACE(IScreen);
+        float pixelRatio = screen->getDevicePixelRatio();
+        
+        auto handle = window->getWindowHandle();
+        const auto &size = window->getViewSize();
+        uint32_t windowId = window->getWindowId();
+
+        gfx::SwapchainInfo info;
+        info.width  = static_cast<uint32_t>(size.x) * pixelRatio;
+        info.height = static_cast<uint32_t>(size.y) * pixelRatio;
+        info.windowHandle = reinterpret_cast<void *>(handle);
+        info.windowId = windowId;
+
+        gfx::Swapchain *swapchain = gfx::Device::getInstance()->createSwapchain(info);
+        _swapchains.emplace_back(swapchain);
+
+        gfx::RenderPassInfo renderPassInfo;
+
+        gfx::ColorAttachment colorAttachment;
+        colorAttachment.format = swapchain->getColorTexture()->getFormat();
+        renderPassInfo.colorAttachments.emplace_back(colorAttachment);
+
+        auto &depthStencilAttachment = renderPassInfo.depthStencilAttachment;
+        depthStencilAttachment.format = swapchain->getDepthStencilTexture()->getFormat();
+        depthStencilAttachment.depthStoreOp = gfx::StoreOp::DISCARD;
+        depthStencilAttachment.stencilStoreOp = gfx::StoreOp::DISCARD;
+
+        scene::IRenderWindowInfo windowInfo;
+        windowInfo.title = StringUtil::format("renderWindow_%d", windowId);
+        windowInfo.width = swapchain->getWidth();
+        windowInfo.height = swapchain->getHeight();
+        windowInfo.renderPassInfo = renderPassInfo;
+        windowInfo.swapchain = swapchain;
+        _mainRenderWindow = createWindow(windowInfo);
```

Comment:
_mainRenderWindow is set in `initialize()`, why assign it again here? And its value is changed when create a system window?